tutor/src/components/homework-questions.js

```diff
@@ -155,34 +163,41 @@ export { Question, QuestionPreview, QuestionHeader, ExerciseNumber };
 
 const ReviewExerciseCard = observer(({
   index, info,
+  showSection,
   questionInfoRenderer: QuestionInfo,
   headerContentRenderer: HeaderContent,
   footerContentRenderer: FooterContent,
   questionType = 'teacher-preview',
   styleVariant = 'points',
-}) => (
-  <QuestionPreview className="openstax-exercise-preview">
-    <QuestionHeader variant={styleVariant} className="question-header">
-      <HeaderContent
-        styleVariant={styleVariant}
-        info={info}
-        label={`Question ${index + 1}`}
-      />
-    </QuestionHeader>
-    <Body className="card-body" variant={styleVariant}>
-      <Question
-        className="openstax-question-preview"
-        question={info.question}
-        hideAnswers={false}
-        choicesEnabled={false}
-        displayFormats={false}
-        type={questionType}
-      />
-      {QuestionInfo && <QuestionInfo info={info} />}
-      {FooterContent && <FooterContent info={info} />}
-    </Body>
-  </QuestionPreview>
-));
+}) => {
+  return (
+    <QuestionPreview className="openstax-exercise-preview">
+      <QuestionHeader variant={styleVariant} className="question-header">
+        <HeaderContent
+          styleVariant={styleVariant}
+          info={info}
+          label={`Question ${index + 1}`}
+        />
+      </QuestionHeader>
+      <Body className="card-body" variant={styleVariant}>
+        <Question
+          className="openstax-question-preview"
+          question={info.question}
+          hideAnswers={false}
+          choicesEnabled={false}
+          displayFormats={false}
+          type={questionType}
+        />
+        {Boolean(showSection &&
+        info.exercise &&
+        info.exercise.page &&
+        info.exercise.page.title) && <a target="_blank" href={`/book/${info.exercise.book.id}/page/${info.exercise.page.id}`}><BookPartTitle part={info.exercise.page} displayChapterSection /></a> }
+        {QuestionInfo && <QuestionInfo info={info} />}
```

Comment:
All the && makes me shudder 😁   How about making a `SectionLink` component that does it and then have this be: `{showSection && <SectionLink info={info} />}` to match the QuestionInfo and FooterContent lines.

We can also use lodash's get, so it could be: `get(info, 'exercise.page')`.  Note I left of title there since if we have a page it should always have a title.

Reply:
:+1: